src/session.ts

```diff
@@ -66,18 +65,16 @@ export class SessionManager implements Middleware {
         vscode.env.sessionId === "someValue.sessionId";
 
     constructor(
-        private requiredEditorServicesVersion: string,
         private log: Logger,
         private documentSelector: DocumentSelector,
-        private hostName: string,
-        private version: string,
-        private reporter: TelemetryReporter) {
+        hostName: string,
+        version: string,
+        private telemetryReporter: TelemetryReporter) {
```

Comment:
Did you want to do this for hostname and version above it too?

I actually started the opposite change in debugAdapter.ts (by using underscore like we do in C#....) but I'm on the fence about it.

I feel conflicted about the inline private declaration in the constructors... It's less boilerplate, but not as discoverable IMO.

I'm fine making it consistent with what is used in the file now, but (eventually...) I may refactor one way or the other.

Reply:
The reason why I removed the private inline field on `hostName` and `version` is because it is not used but rather directly assigned to the backing fields `HostName` and `HostVersion` and those properties get later used/modified. We should probably rename the `version` parameter to `hostVersion` if you agree? In terms of casing, I quite like the Upper casing and wouldn't use underscores TBH. The official `TypeScript` guide explicitly recommends against that : https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines
> "Do not use "_" as a prefix for private properties."

JavaScript doesn't have a notion of private/public, therefore the private modified in TypeScript is just some syntactic sugar to give a compilation error if the stated intent is not followed, hence why people probably only have either a public or private field and therefore don't need a prefix it seems.

The reason why I didn't remove the private inline field it for `telemetryReporter` is because that private property is actually being used at the moment. Due to the `this.` accessor, the reader knows that it's a property and not local variable/parameter. To me, using the private auto-created property signifies that we use the passed in parameter directly as-is and not a potentially modified, cached version (yes, one could've technically modify that property but the lower case shows signals to me the intent of just using the passed in parameter as-is, maybe that's just me).
Happy to go for whatever you prefer, if I read correctly, you'd prefer to create an explicit backing field `telemetryReporter`, right?

Reply:
It would be nice to eventually get consistent on whether we use ctror inline private (or public/protected) declarations.  I lean towards using them since they do reduce boilerplate code and less code means less opportunity for bugs to creep in.